tests/admin_changelist/tests.py

```diff
@@ -761,6 +762,14 @@ def test_pagination_page_range(self):
                 list(real_page_range),
             )
 
+    def test_object_tools_displayed_no_add_permission(self):
+        "When has_add_permission is False, the object tools block is still shown."
+        superuser = self._create_superuser('superuser')
+        m = EventAdmin(Event, custom_site)
+        request = self._mocked_authenticated_request('/event/', superuser)
+        response = m.changelist_view(request)
+        self.assertIn('<ul class="object-tools">', response.rendered_content)
```

Comment:
should we also assert that "Add {{ name }}" isn't shown? If there's already a test for that, maybe we could simply modify that one?